analyzer/analyzer_test.go

```diff
@@ -176,7 +177,10 @@ var rulesFile = %s
 			if err != nil {
 				t.Fatal(err)
 			}
-			defer os.Remove(mainFile.Name())
+			defer func() {
+				mainFile.Close()
+				os.Remove(mainFile.Name())
+			}()
```

Comment:
To be honest, we can rewrite it with https://pkg.go.dev/testing#T.TempDir and just drop folder after a test. However this can be less important than just close a file.